sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -1679,12 +1672,6 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase with Compilat
       messageParameters = Map.empty)
   }
 
-  def cannotSaveVariantIntoExternalStorageError(): Throwable = {
-    new AnalysisException(
-      errorClass = "CANNOT_SAVE_VARIANT",
-      messageParameters = Map.empty)
```

Comment:
Are these two error message templates, `CANNOT_SAVE_VARIANT` and `INDETERMINATE_COLLATION`, still needed?

Reply:
`CollationIdentifier`? `INDETERMINATE_COLLATION_ID`?

Reply:
Yes，if safe I remove INDETERMINATE_COLLATION_ID from error-conditions.json @LuciferYang

Reply:
OK，let's remove INDETERMINATE_COLLATION from error-conditions.json @LuciferYang

Reply:
Done